src/seedu/addressbook/data/person/Address.java

```diff
@@ -25,34 +31,151 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(address)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = address;
+        createAddress(address);
+    }
+
+    private void createAddress(String address) throws IllegalValueException{
+        final int INDEX_BLOCK_ADDRESS = 0;
+        final int INDEX_STREET_ADDRESS = 1;
+        final int INDEX_UNIT_ADDRESS = 2;
+        final int INDEX_POSTCODE_ADDRESS = 3;
+        String[] addressSegment = address.split(",");
+        if (addressSegment.length < 4){
+            throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
+        }
```

Comment:
You have done the validation in the above method, no need to do same thing again here.